docs/how-to/h-external-k8s-connection.md

```diff
@@ -0,0 +1,72 @@
+    - Supported authentication mechanisms include `SCRAM-SHA-512`, `OAUTHBEARER` and `SSL`
+- The remaining NodePort services each correspond to a specific unit-id, security protocol and authentication mechanism, with fixed Ports and randomly allocated NodePorts
+
+
+> **Note** - The `kafka-k8s` and `kafka-k8s-endpoints` ClusterIP services seen above are created for every Juju application by default as part of the StatefulSet they are associated with. Generally, these services are not relevant to users and can be safely ignored
+
+## Setting up NodePort services
+
+By default, the `kafka-k8s-bootstrap` NodePort service will be created automatically, however the `NodeIP:NodePort` addresses will be initially unreachable.
+
+To connect to Charmed Kafka K8s use Kubernetes-external clients, the aforementioned services must be associated with individual listeners. To achieve this:
+1. Set the charm config option `expose-external=nodeport` on the Kafka K8s application
+    - This instructs each broker unit to create it's own individual NodePort services
+    - By default this value is `false`
+2. Enable at least one listener using the appropriate relation endpoints
+    - Each security protocol + authentication mechanism corresponds to specific port numbers
+    - For additional reference on how to enable various security protocols + authentication mechanisms and their associated ports, please refer to [Charmed Kafka K8s Documentation - Reference Listeners](/t/charmed-kafka-k8s-documentation-reference-listeners/13270)
+
+Once both conditions have been met, the Charmed Kafka K8s application will expose the per-unit NodePort services. Examples of how these look - for example the `kafka-k8s-0-sasl-plaintext-scram` service - can be found in the table in the [previous section](#external-k8s-connection).
+
+## Client connections using the bootstrap service
+
+A client can be configured to connect to the `kafka-k8s-bootstrap` service using a Kubernetes NodeIP, and desired NodePort.
+
+To get NodeIPs:
+
+```console
+$ kubectl get nodes -o wide -n model | awk -v OFS='\t\t' '{print $1, $6}'
+
+NAME        INTERNAL-IP
+node-0      10.155.67.110
+node-1      10.155.67.120
+node-2      10.155.67.130
+```
+
+To get bootstrap NodePorts, if using the example from the [previous section](#external-k8s-connection), they would be:
+```shell
+29092:31982/TCP,29093:31262/TCP,29094:30845/TCP,29095:30153/TCP,29096:32601/TCP
+```
+
+> **Note** - The values shown here for NodeIP and NodePort are just examples. NodeIPs will be different for each Kubernetes cluster, and NodePorts are randomly allocated for each deployment
+
+
+When a client application attempts to connect to the `kafka-k8s-bootstrap` service, for example on `node-0:31262` for a `SASL_SSL` security protocol + `SCRAM-SHA-512` authentication mechanism endpoint, Kubernetes will route that request round-robin to one of the available Pods (units) within the `kafka-k8s` StatefulSet (application).
```

Comment:
**thought** the part below seems `Explanation` to me. Maybe I would remove this, and possibly write an Explanation on how ingress works on Kafka. Maybe we can shrink the how-to guide here to the bare minimum, with what to do to enable ingress. 

Indeed to set up ingress we need to tell the users two things:
* how-to set a config option 
* how to fetch the correct host-ip and node port

All the rest, should probably be explanation, together with the actual examples here. 

Anyhow, let's see what @izmalk says

Reply:
I like the idea of cutting down the How to guide to a bare minimum (practical step-by-step instructions on how to do stuff). Maybe later we can include the rest on a page in the Explanation section and provide the link here.

Reply:
I'll nix L66+ in that case! No problem.